File: app/src/main/java/sb/yoon/kiosk2/ResultActivity.java
package sb.yoon.kiosk2;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import java.util.Set;

public class ResultActivity extends Activity {
	private TextView result;
	public String msg = "";
	public Intent i;
	private Bundle extras;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_result);
		result = (TextView)findViewById(R.id.TextView1);
		i = getIntent();
		extras = i.getExtras();
		String card = (String) extras.get("ACQUIRER_NAME");
		String result_code = (String) extras.get("RESULT_CODE");
		String total_amount = (String) extras.get("TOTAL_AMOUNT");
		printInent(i);

		if (result_code.equals("0000")) {
			Log.d("결제완료", card + "로 " + total_amount + "만큼 " + "결제되었습니다.");
		}
	}

	public void printInent(Intent i) {
		try {
			//Log.e("KTC","-------------------------------------------------------");
			//util.saveLog("-------------------------------------------------------");
			if (i != null) {
				if (extras != null) {
					Set keys = extras.keySet();

					for (String _key : extras.keySet()) {
						Log.e("RESULTACT","key=" + _key + " : " + extras.get(_key));
						msg += "\n" + _key + "=";
						msg += extras.get(_key);
					}
				}
				// result.setText(msg);
			}
		} catch (Exception e) {
			e.printStackTrace();
		} finally {
		}
	}

	public void onBackPressed() {
		finishAffinity();
	}

}

File: app/src/main/java/sb/yoon/kiosk2/ResultActivity.kt
package sb.yoon.kiosk2

import android.app.Activity
import android.widget.TextView
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import java.lang.Exception

class ResultActivity : Activity() {
    private var result: TextView? = null
    var msg: String? = ""
    private var extras: Bundle? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_result)
        result = findViewById<View>(R.id.TextView1) as TextView
        extras = intent.extras
        val card = extras!!["ACQUIRER_NAME"] as String?
        val resultCode = extras!!["RESULT_CODE"] as String?
        val totalAmount = extras!!["TOTAL_AMOUNT"] as String?
        printInent(intent)
        if (resultCode == "0000") {
            Log.d("결제완료", card + "로 " + totalAmount + "만큼 " + "결제되었습니다.")
        }
    }

    fun printInent(i: Intent?) {
        try {
            if (i != null) {
                if (extras != null) {
                    val keys: Set<*> = extras!!.keySet()
                    for (_key in extras!!.keySet()) {
                        Log.e("RESULTACT", "key=" + _key + " : " + extras!![_key])
                        msg += "\n$_key="
                        msg += extras!![_key]
                    }
                }
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun onBackPressed() {
        finishAffinity()
    }
}